packages/local-notifications/index.ios.ts

```diff
@@ -235,48 +256,44 @@ export class LocalNotificationsImpl extends LocalNotificationsCommon implements
 		return cal;
 	}
 
-	private static schedulePendingNotificationsLegacy(pending: ScheduleOptions[]): Array<number> {
-		const scheduledIds: Array<number> = [];
-		for (const n in pending) {
-			const options = LocalNotificationsImpl.merge(pending[n], LocalNotificationsImpl.defaults);
+	private static schedulePendingNotificationsLegacy(scheduleOptions: ScheduleOptions[]): Array<number> {
+		const scheduledIds: number[] = [];
+
+		for (const s of scheduleOptions) {
+			const entry = LocalNotificationsImpl.createScheduleEntry(s);
 
-			LocalNotificationsImpl.ensureID(options);
-			scheduledIds.push(options.id);
+			scheduledIds.push(entry.id);
 
 			const notification = UILocalNotification.new();
-			notification.fireDate = options.at ? options.at : new Date();
-			notification.alertTitle = options.title;
-			notification.alertBody = options.body;
+			notification.fireDate = entry.at ? entry.at : new Date();
+			notification.alertTitle = entry.title;
+			notification.alertBody = entry.body;
 
 			notification.timeZone = NSTimeZone.defaultTimeZone;
-			notification.applicationIconBadgeNumber = options.badge;
+			notification.applicationIconBadgeNumber = entry.badge;
 
 			// these are sent back to the plugin when a notification is received
 			const userInfoDict = NSMutableDictionary.alloc().initWithCapacity(4);
-			userInfoDict.setObjectForKey(options.id, 'id');
-			userInfoDict.setObjectForKey(options.title, 'title');
-			userInfoDict.setObjectForKey(options.body, 'body');
-			userInfoDict.setObjectForKey(options.interval, 'interval');
-			if ('payload' in options) {
-				userInfoDict.setObjectForKey(JSON.stringify(options.payload), 'payload');
+			userInfoDict.setObjectForKey(entry.id, 'id');
+			userInfoDict.setObjectForKey(entry.title, 'title');
+			userInfoDict.setObjectForKey(entry.body, 'body');
+			userInfoDict.setObjectForKey(entry.interval, 'interval');
+			if ('payload' in entry) {
+				userInfoDict.setObjectForKey(JSON.stringify(entry.payload), 'payload');
 			}
 			notification.userInfo = userInfoDict;
 
-			switch (options.sound) {
-				case null:
-				case false:
-					break;
-				case undefined:
-				case 'default':
+			if (entry.sound) {
```

Comment:
Fold `undefined` old logic would use default sound, unlike in the rewritten logic. So this should be adjusted to account for no sound sent.

Reply:
Reverted to using the good old switch statement with the addition of `''` check.